src/app/modules/shared_models/collectibles_model.nim

```diff
@@ -262,6 +283,44 @@ QtObject:
     self.endRemoveRows()
     self.countChanged()
 
+  proc updateCollectibleItems(self: Model, newItems: seq[CollectiblesEntry]) =
+    if len(self.items) == 0:
+      self.resetCollectibleItems(newItems)
+      return
+    
+    if len(newItems) == 0:
+      self.removeCollectibleItems()
+      return
+
+    var newTable = initTable[string, int](len(newItems))
+    for i in 0 ..< len(newItems):
+      newTable.add(newItems[i].getID(), i)
+
+    # Needs to be built in sequential index order
+    var oldIndicesToRemove: seq[int] = @[]
+    for idx in 0 ..< len(self.items):
+      let uid = self.items[idx].getID()
+      if not newTable.hasKey(uid):
+        # Item in old list but not in new -> Must remove
+        oldIndicesToRemove.add(idx)
+      else:
+        # Item both in old and new lists -> Nothing to do
```

Comment:
That comment doesn't seem to match the delete below

Reply:
rewrote comment